src/include/sof/audio/component_ext.h

```diff
@@ -171,35 +171,8 @@ static inline int comp_prepare(struct comp_dev *dev)
 	return 0;
 }
 
-/** See comp_ops::copy */
-static inline int comp_copy(struct comp_dev *dev)
-{
-	int ret = 0;
-
-	assert(dev->drv->ops.copy);
-
-	/* copy only if we are the owner of the component */
-	if (cpu_is_me(dev->ipc_config.core)) {
-/* BugLink: https://github.com/zephyrproject-rtos/zephyr/issues/43786
- * TODO: Remove this once the bug gets fixed.
- */
-#ifndef __ZEPHYR__
-		perf_cnt_init(&dev->pcd);
-#endif
-
-		ret = dev->drv->ops.copy(dev);
+int comp_copy(struct comp_dev *dev);
```

Comment:
I think this should be in component.h

Reply:
there is no component_ext.c for source file, so component.c can cover component_ext.h and component.h both.
and if move to component.h will cause much more changes, a lot of component_ext.h include will change to component.h.
Agree?

Reply:
component_ext.h contains all inline functions, no function prototype in it. while component.c and component.h are .c/.h pair.

Reply:
I searched #include <sof/audio/component_ext.h> under SOF directory, it have nearly 50 place to include, if move it to component.h, it will cause much more source file to change.

I don't know history about component.h and compont_ext.h, but currently, they have same component.c source file.
even check component.h, it is mixed with macro, inline, prototype.

If we really want to change, I would suggest discuss first, then raise an CR to merge(or re-arrange) two header files

Agree?

Reply:
> it have nearly 50 place to include, if move it to component.h, it will cause much more source file to change.

This would not be pleasant but it's not the end of world either: it's only a one-line change in each file so this would not break `git` much. If this is really the best choice then don't be afraid to do it. 

Most projects have some messy `#include`s that no one really cares about... until someone tries to move something exactly like as you're doing here; it's typical. Sometimes this can trigger some `#include` cleanup and that's usually a good thing.

Note: I do NOT know whether it's the best choice! Discuss and make sure first. Considering how longer and longer this thread is getting, a call should help.

Reply:
@btian1 I think keeping the comp_copy() declaration is component_ext.h is fine. All the other ops' declaration or definition are in this file. So I see no objection here

Reply:
> I don't know history about component.h and compont_ext.h

Github's browsing features are very user-friendly. Sometimes I drop `tig` and switch to the web browser instead!

`...` (top right corner) -> View File -> Blame -> line 1 ->

```
comp: api: advanced and internal functions separated

The current content of component.h is a mix of basic api, common
basic helpers for every component developers as well as advanced
functions and macros used by infrastructure and very specialized
components like host, dai, kpb etc.

This patch moves the advanced part to component_ext.h and keeps
only basic part in component.h to avoid overloading effects
component developers with declarations and code they do not use.
```

Reply:
> I think this should be in component.h

No, it should not. It is a little helper function that hides comp driver/device details, called by the infrastructure. While the component.h was intended to declare things relevant for component implementors, see the commit message @marc-hb mentioned in the comment above. component.h had been overloaded and unreadable before component_ext.h was created to separate internal non-api things. Not sure if merging back is a good idea.